keps/sig-autoscaling/1610-container-resource-autoscaling/README.md

```diff
@@ -281,19 +439,594 @@ unit tests will be added:
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+The feature can be disabled in Alpha and Beta versions
+by restarting kube-apiserver and kube-controller-manager with the feature-gate off.
+In terms of Stable versions, users can choose to opt-out by not setting the
+`ContainerResource` type metric in their HPA.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+HPA with `ContainerResource` type metric can be created and can be handled by HPA controller.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+No. But, the tests to confirm the behavior on switching the feature gate will be added.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+When a rollout fail, it shouldn't impact already running HPAs because it's an opt-in feature,
+and users need to set `ContainerResource` metric to use this feature. 
+
+When a rollback fail for kube-controller-manager, HPA controller will continue to handle `ContainerResource` metric in HPAs.
+When a rollback fail for kube-apiserver, but success kube-controller-manager, 
+HPA controller will just ignore `ContainerResource` metric in HPAs.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+N/A
```

Comment:
This shouldn't be N/A for beta.

Reply:
Agreed. Ideally this is metrics, but even if it's "scrape the API for HPA failures", we need to provide some guidance to folks. As I state in another issue, I think as a condition of approval here I would like HPA metrics added to the beta criteria.

Reply:
changed here to refer to the metrics proposed in "What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?" section.